src/traction_modem/MemorySpace.hxx

```diff
@@ -96,9 +97,20 @@ public:
             len = Defs::MAX_WRITE_DATA_LEN;
         }
         // Register for a write response and send the read request.
-        rxFlow_->register_handler(this, Defs::RESP_MEM_W);
-        txFlow_->send_packet(Defs::get_memw_payload(
-            get_space_id(), destination, data, len));
+        link_->get_rx_iface()->register_handler(this, Defs::RESP_MEM_W);
+        Defs::Payload p =
+            Defs::get_memw_payload(get_space_id(), destination, data, len);
+        if (link_->is_link_up())
+        {
+            // Link is up, we can send the write request immediately.
+            link_->get_tx_iface()->send_packet(p);
+        }
+        else
+        {
+            // Link is down, let's save the message in case the link comes up
```

Comment:
what is the reason for adding this complexity here? Why do we need to handle memory config requests when there is no link?
Normally the link doesn't just come and go. It should come up at boot and that's it.

The reason for asking is that it feels like there is a lyering problem here. This is an application level protocol. This layer should not be worried about what is going on with the link. If it does, we have to copy-paste this code to N application level protocol handlers that we have. So if there is a legit feature here, that feature hsould not exist at this layer but at a lower layer so that we on't have to keep repeating this code (and these tests) over and over again.